Tests/ApolloTests/MutatingSelectionSetTests.swift

```diff
@@ -86,4 +86,153 @@ class MutatingSelectionSetTests: XCTestCase {
     expect(data2.hero.name).to(equal("Leia"))
   }
 
+  func test__mutateIfFulfilled_typeCaseProperties__typeCaseIsFulfilled_mutatesProperties() throws {
+    // given
+    var model = TypeCaseSelectionSet.AsHero(__typename: "Hero", name: "A", age: 1).asRootEntityType
+
+    // when
+    let didMutate = model.mutateIfFulfilled(\.asHero) { hero in
+      hero.name = "B"
+      hero.age = 2
+    }
+
+    // then
+    expect(didMutate).to(beTrue())
+    expect(model.asHero?.name).to(equal("B"))
+    expect(model.asHero?.age).to(equal(2))
+  }
+
+  func test__mutateIfFulfilled_typeCaseProperty__typeCaseIsNotFulfilled_returnsFalseAndDoesNotCallMutationBlock() throws {
+    // given
+    var model = TypeCaseSelectionSet(__typename: "Hero")
+
+    // when
+    let didMutate = model.mutateIfFulfilled(\.asHero) { hero in
+      fail("Should not call mutation block")
+    }
+
+    // then
+    expect(didMutate).to(beFalse())
+    expect(model.asHero).to(beNil())
+  }
+
+  func test__mutateIfFulfilled_doubleNestedTypeCaseProperties__typeCaseIsFulfilled_mutatesProperties() throws {
+    // given
+    var model = TypeCaseSelectionSet.AsHero.AsHuman(
+      __typename: "Hero", name: "A", age: 1, birthMonth: "Jan"
+    ).asRootEntityType
+
+    // when
+    let didMutate = model.mutateIfFulfilled(\.asHero?.asHuman) { hero in
+      hero.birthMonth = "Feb"
+    }
+
+    // then
+    expect(didMutate).to(beTrue())
+    expect(model.asHero?.asHuman?.birthMonth).to(equal("Feb"))
+  }
+
+  // MARK: - Shared Models
+
+  struct TypeCaseSelectionSet: MockMutableRootSelectionSet {
+    public var __data: DataDict = .empty()
+    init(_dataDict: DataDict) { __data = _dataDict }
+
+    static var __selections: [Selection] { [
+      .inlineFragment(AsHero.self)
+    ]}
+
+    var asHero: AsHero? { _asInlineFragment() }
+
+    init(
+      __typename: String
+    ) {
+      self.init(_dataDict: DataDict(
+        data: [
+          "__typename": __typename
+        ], fulfilledFragments: [
+          ObjectIdentifier(Self.self),
+        ]))
+    }
+
+    struct AsHero: MockMutableInlineFragment {
+      typealias RootEntityType = TypeCaseSelectionSet
+
+      static var __parentType: any ApolloAPI.ParentType { Object.mock }
+
+      public var __data: DataDict = .empty()
+      init(_dataDict: DataDict) { __data = _dataDict }
+
+      static var __selections: [Selection] { [
```

Comment:
I don't think it matters to this particular test suite but it looks like we're missing the `age` selection.